docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,598 @@
+
+## Introduction
+
+Pydantic began life as an experiment in some code in a long dead project.
+I ended up making the code into a package and releasing it.
+It got a bit of attention on hacker news when it was first released, but started to get really popular when
+Sebastián Ramírez used it in [FastAPI](https://fastapi.tiangolo.com/).
+Since then the package and its usage have grown enormously.
+The core logic however has remained relatively unchanged since the initial experiment.
+It's old, it smells, it needs to be rebuilt.
+
+The release of version 2 is an opportunity to rebuild pydantic and correct many things that don't make sense - 
+**to make pydantic amazing :rocket:**.
+
+Much of the work on V2 is already done, but there's still a lot to do.
+Now seems a good opportunity to explain what V2 is going to look like and get feedback from users.
+
+## Headlines
+
+For good and bad, here are some of the biggest changes expected in V2.
+
+The core validation logic of pydantic V2 will be performed by a separate package 
+[pydantic-core](https://github.com/samuelcolvin/pydantic-core) which I've been building over the last few months.
+
+*pydantic-core* is written in Rust using the excellent [pyo3](https://pyo3.rs/) library which provides rust bindings
+for python.
+
+!!! note
+    The python interface to pydantic shouldn't change as a result of using pydantic-core, instead
+    pydantic will use type annotations to build a schema for pydantic-core to use.
+
+pydantic-core is usable now, albeit with a fairly unintuitive API, if you're interested, please give it a try.
+
+pydantic-core provides validators for all common data types, 
+[see a list here](https://github.com/samuelcolvin/pydantic-core/blob/main/pydantic_core/_types.py#L291).
+Other, less commonly used data types will be supported via validator functions.
+
+### Performance :thumbsup:
+
+As a result of the move to rust for the validation logic 
+(and significant improvements in how validation objects are structured) pydantic V2 will be significantly faster
+than pydantic V1.
+
+Looking at the pydantic-core [benchmarks](https://github.com/samuelcolvin/pydantic-core/tree/main/tests/benchmarks)
+today, pydantic V2 is between 4x and 50x faster than pydantic V1.9.1.
+
+In general, pydantic V2 is about 17x faster than V1 when validating a model containing a range
+of common fields.
+
+### Strict Mode :thumbsup:
+
+People have long complained about pydantic for coercing data instead of throwing an error.
+E.g. input to an `int` field could be `123` or the string `"123"` which would be converted to `123`.
+
+pydantic-core comes with "strict mode" built in. With this only the exact data type is allowed, e.g. passing
+`"123"` to an `int` field would result in a validation error.
+
+This will allow pydantic V2 to offer a `strict` switch which can be set on either a model or a field.
+
+#### `IsInstance` checks :thumbsup:
```

Comment:
That will be very nice even for tests (this + dirty-equals should make shape testing super easy)